macros/cross_db_utils/date_trunc.sql

```diff
@@ -7,5 +7,5 @@
 {% endmacro %}
 
 {% macro bigquery__date_trunc(datepart, date) %}
-    date_trunc({{date}}, {{datepart}})
+    date_trunc(cast({{date}} as date), {{datepart}})
```

Comment:
do you remember why you made this change? If we cast the argument to a `date`, then it will lose its hours/mins/seconds. If you call this with:
```
{{ date_trunc("'2018-01-01 12:34:56'", "hour") }}
```

I think the result will be `2018-01-01` instead of `2018-01-01 12:00:00`. Can we cast this to a `timestamp` and use `timestamp_trunc` instead?

Reply:
so in the .csv file, it looks like the expected result is looking for just the date without the timestamp associated with it. if i cast the argument to `timestamp_trunc` it's going to look for the some timestamp. that's why i cast the argument to `date_trunc`....

Reply:
using the `timestamp_trunc` function throws this error: `No matching signature for operator != for argument types: TIMESTAMP, DATE` when i try to test `test_date_trunc`